vktrace/vktrace_common/vktrace_filelike.c

```diff
@@ -88,14 +88,57 @@ uint64_t vktrace_FileLike_GetFileLength(FILE* fp) {
 }
 
 // ------------------------------------------------------------------------------------------------
-FileLike* vktrace_FileLike_create_file(FILE* fp) {
+FileLike* vktrace_FileLike_create_file(FILE* fp, BOOL preload) {
     FileLike* pFile = NULL;
     if (fp != NULL) {
         pFile = VKTRACE_NEW(FileLike);
-        pFile->mMode = File;
-        pFile->mFile = fp;
-        pFile->mMessageStream = NULL;
+        pFile->mFile = NULL;
         pFile->mFileLen = vktrace_FileLike_GetFileLength(fp);
+        pFile->mMessageStream = NULL;
+        pFile->mMemAddr = NULL;
+        pFile->mMemCurrAddr = NULL;
+
+        if (pFile->mFileLen == 0) {
+            vktrace_LogError("Failed to create FileLike, file length is 0!");
```

Comment:
Avoid using a term like "FileLike". This is a concept internal to the vkreplay code, give an error message that someone that isn't familiar with vkreplay internals can use to identify the problem.

Reply:
Sure.  That makes sense, will update the message to "Failed to read file, file length is 0!"